src/Datagrid/ProxyQuery.php

```diff
@@ -202,6 +202,13 @@ public function setSortBy(array $parentAssociationMappings, array $fieldMapping)
         return $this;
     }
 
+    public function resetSortBy(): BaseProxyQueryInterface
```

Comment:
Do you plan to add this to the ProxyQueryInterface interface ?

Reply:
I guess we will need to... unless there is another way to accomplish the same effect. getDoctrineQuery uses the sortBy to add more sorts to the final query. Is there another workaround that avoids this new method?

Reply:
I see some issues with this:
- If we're introducing it to SonataAdmin/ProxyQueryInterface, this is a method which is not needed for other ORM
- If we're introducing it to SonataORM/ProxyQueryInterface, it just makes the interface bigger and bigger (See https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/4.x/src/Datagrid/ProxyQueryInterface.php vs https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/blob/4.x/src/Datagrid/ProxyQueryInterface.php)
- If we introduce this, there should be a `resetSortOrder` too to be consistent
- One concern is the fact that `$query->setOrderBy(...)->resetOrderBy()` is doing something under the hood: It add a LEFT JOIN in the query https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/4.x/src/Datagrid/ProxyQuery.php#L199 when you could believe the reset does cancel the full setOrderBy call.

The only other option I see would be to change the `getDoctrineQuery` call depends on the reason we have to remove the sort by:
- If this is a temporary solution until some support is added to Doctrine, we could use `$query->getQueryBuild()->setSortBy(null)->getQuery()` in the case of OneToMany...
- If this is a long-term solution cause no support will be added we could consider changing `getDoctrineQuery` to `getDoctrineQuery(bool $withSort = true)`.

Also, why I don't understand between this PR and the issue https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1701 is the fact it was said that before this fix (https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1639) it was working. But the quoted fix didn't add any sortBy, just a select. So why the `$query->resetSortBy()` is needed (or was it really working before) ?

Reply:
About the "how to deal with the sort by" I will think about all the points you mentioned.

And about the why is this needed. I will introduce a commit reverting the fix, so the test fail and we can see the error message (just in case there is a better fix). The issue is present only on one to many. And I think the test cover all possible sort by options (property on the class, relation oto, otm, mtm, mto)

Reply:
> And about the why is this needed. I will introduce a commit reverting the fix, so the test fail and we can see the error message (just in case there is a better fix). The issue is present only on one to many. And I think the test cover all possible sort by options (property on the class, relation oto, otm, mtm, mto)

Thanks.

one thing to try would be
```
if ($rootAliasSortBy !== $rootAlias && !$this->isOneToMany($query, $rootAliasSortBy)) {
    $query->getQueryBuilder()->addSelect($rootAliasSortBy);
}
```
According do the message https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1701#issuecomment-1293203154 with the working and not working query.

Reply:
Thats what I am trying now, seems that removing the sort is not needed, just not add the select.

Reply:
I tried another approach, lets see if it is good enough